loadtest/store/memstore/store.go

```diff
@@ -974,3 +985,123 @@ func (s *MemStore) SetServerVersion(version string) error {
 	s.serverVersion = version
 	return nil
 }
+
+// SetThread stores the given thread reponse.
+func (s *MemStore) SetThread(thread *model.ThreadResponse) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	if thread == nil {
+		return errors.New("memstore: thread should not be nil")
+	}
+
+	// We get an element from the queue and check if we have it in the map and
+	// if it points to the same memory location. If so, we delete it since it means the queue is full.
+	// This is done to keep the data pointed by the map consistent with the data stored in the queue.
+	t := s.threadsQueue.Get().(*model.ThreadResponse)
+	if tt, ok := s.threads[t.PostId]; ok && tt == t {
+		delete(s.threads, t.PostId)
+	}
+	cloneThreadResponse(thread, t)
+	s.threads[thread.PostId] = t
+
+	return nil
+}
+
+// SetThreads stores the given thread response as a thread.
+func (s *MemStore) SetThreads(trs []*model.ThreadResponse) error {
+	if len(trs) == 0 {
+		return nil
+	}
+	for _, tr := range trs {
+		if err := s.SetThread(tr); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *MemStore) getThreads(unreadOnly bool) ([]*model.ThreadResponse, error) {
+	var threads []*model.ThreadResponse
+	for _, thread := range s.threads {
+		if unreadOnly && thread.UnreadReplies == 0 {
+			continue
+		}
+		threads = append(threads, cloneThreadResponse(thread, &model.ThreadResponse{}))
+	}
+	return threads, nil
+}
+
+// ThreadsSorted returns all threads, sorted by LastReplyAt
+func (s *MemStore) ThreadsSorted(unreadOnly, asc bool) ([]*model.ThreadResponse, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	threads, err := s.getThreads(unreadOnly)
```

Comment:
Unless necessary I think we should unlock as soon as we get the threads and do the sorting outside.